### ISSUE TITLE
Assembly.CreateInstance returning null instead of throwing an ArgumentException

### ISSUE THREAD

**Issue by non-conformances-research** [dotnet/corefx#35857]
According to the specification, Assembly.CreateInstance method should throw an ArgumentException to the above program.

```
using System;
using System.Reflection;

public class A {
  public static void Main() {
    Assembly assem = typeof(A).Assembly;
    A a = (A)assem.CreateInstance(" ");
    if (a == null) {
      Console.WriteLine("Null");
    } else {
      Console.WriteLine(a);
    }
  }
}
```

## Steps to Reproduce

1. Compile above program.
2. Run it.

### Current Behavior

Null

### Expected Behavior

ArgumentException thrown.

## On which platforms did you notice this

[ ] macOS
[X] Linux
[ ] Windows

**Version Used**:

.NET Core SDK 2.1.401

**Comment by jkotas** [comment]
> According to the specification

Could you please share link to the specification that you are using? The documentation at https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.createinstance?view=netcore-2.1  says:

    If the runtime is unable to find typeName in the Assembly instance, it returns null instead of throwing an exception

So this behaves exactly as documented.

**Comment by non-conformances-research** [comment]
Why not throw an ArgumentException like when we pass ""? What is the difference?

**Comment by MichalStrehovsky** [comment]
Empty type name is not allowed per ECMA-335 II.22.37 (TypeDef).

Type name that is a space "` `" is a valid type name.

**Comment by jkotas** [comment]
> Type name that is a space `" "` is a valid type name.

That's right. A small wrinkle is that `Assembly.GetType` or `Assembly.CreateInstance` does not actually work for a type with `" "` name, e.g try this:

```csharp
using System;
using System.Reflection;
using System.Reflection.Emit;

class Program
{
    static void Main(string[] args)
    {
        string typeName = "A"; // Works fine 

        var ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("MyAssembly"), AssemblyBuilderAccess.Run);
        var mb = ab.DefineDynamicModule("MyModule");
        var tb = mb.DefineType(typeName);
        var t = tb.CreateType();

        Console.WriteLine((t.Assembly.GetType(typeName) != null) ? "Found" : "Not found");
    }
}
```

**Comment by steveharter** [comment]
I recommend closing this as this API (and other similar APIs) don't treat whitespace the same as empty\null.
